src/libxtp/customtools.cc

```diff
@@ -0,0 +1,151 @@
+#include <fstream>
+#include <iostream>
+#include <string.h>
+#include <votca/xtp/customtools.h>
+
+namespace votca {
+namespace xtp {
+
+/* Custom Tools */
+
+void CustomTools::ExportMat(std::string filename, const Eigen::MatrixXd& mat) {
+  Eigen::IOFormat fmt(Eigen::StreamPrecision, 0, ", ", "\n", "", "");
+  std::ofstream file;
+  file.open(filename, std::ios_base::trunc);
+  file << mat.format(fmt) << std::endl;
+  file.close();
+}
+
+void CustomTools::ExportMatBinary(std::string filename, const Eigen::MatrixXd& mat) {
+  std::ofstream out(filename, std::ios::out | std::ios::binary | std::ios::trunc);
+  Eigen::MatrixXd::Index rows = mat.rows(), cols = mat.cols();
+  out.write((char*) (&rows), sizeof(Eigen::MatrixXd::Index));
+  out.write((char*) (&cols), sizeof(Eigen::MatrixXd::Index));
+  out.write((char*) mat.data(), rows * cols * sizeof(Eigen::MatrixXd::Scalar));
+  out.close();
+}
+
+Eigen::MatrixXd CustomTools::ImportMatBinary(std::string filename) {
+  std::ifstream in(filename, std::ios::in | std::ios::binary);
+  Eigen::MatrixXd::Index rows = 0, cols = 0;
+  in.read((char*) (&rows), sizeof(typename Eigen::MatrixXd::Index));
+  in.read((char*) (&cols), sizeof(typename Eigen::MatrixXd::Index));
+  Eigen::MatrixXd mat;
+  mat.resize(rows, cols);
+  in.read((char*) mat.data(), rows * cols * sizeof(Eigen::MatrixXd::Scalar));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).  ](https://app.codacy.com/app/votca-package/xtp/pullRequest?prid=4286871)